### ISSUE TITLE
Rule "handle-callback-err" - Missing callbacks throwing undefined error

### ISSUE THREAD

**Issue by mstuart** [eslint/eslint#740]
I'm getting this error when running grunt-eslint (grabs eslint@0.4.5) ->

``` shell
Running "eslint:server" (eslint) task
Warning: Cannot set property 'handled' of undefined Use --force to continue.

Aborted due to warnings.

Process finished with exit code 3
```

My mocha test is causing this error.  I'm creating an Error object without an err callback. ->

``` javascript
it('should log some errors', function (done) {
   // create a mock error
   var err = new Error('your hair is on fire');
   ...
});
```

Seems that we need to check if callbacks exist before setting `handled` to true.
https://github.com/eslint/eslint/blob/master/lib/rules/handle-callback-err.js#L76

I'll follow up with a pull request w/ a patch.  I'll help with getting this merged ASAP since grunt-eslint is pulling all minor patches from eslint.  Some other folks may get this error too.

/cc @xjamundx @nzakas for comments
